bin/mv.js

```diff
@@ -53,83 +53,80 @@ function handleWebtaskMove(args) {
         throw Cli.error.invalid('Webtasks are identical. Use a different target name, container or profile.');
     }
 
-    return read(profile, name)
-        .then(function(webtask) {
-            return copy(profile, webtask, target);
-        })
-        .then(function() {
-            return profile.removeWebtask({
-                name: name
-            });
-        });
-}
-
-function equal(source, target) {
-    return _.isEqual(source, {
-        name: target.name,
-        container: target.container || source.container,
-        profile: target.profile || source.profile
-    });
-}
-
-function read(profile, name) {
-    return profile.inspectWebtask({
-            name: name,
-            decrypt: true,
-            fetch_code: true
+    return profile.getWebtask({
+            name: name
         })
         .catch(function(err) {
             if (err.statusCode === 404) {
                 throw Cli.error.notFound('No such webtask: ' + Chalk.bold(name));
             }
             throw err;
+        })
+        .then(function(webtask) {
+            sourceWebtask = webtask;
+            return webtask.inspect({
+                decrypt: true,
+                fetch_code: true
+            });
+        })
+        .then(function(data) {
+            return copy(sourceWebtask, data, target);
+        })
+        .then(function() {
+            return sourceWebtask.remove();
         });
 }
 
-function copy(profile, webtask, target) {
-    if (!webtask.jtn) {
+function equal(sourceParams, targetParams) {
+    return _.isEqual(sourceParams, {
+        name: targetParams.name,
+        container: targetParams.container || sourceParams.container,
+        profile: targetParams.profile || sourceParams.profile
+    });
+}
+
+function copy(webtask, data, target) {
+    if (!data.jtn) {
         throw Cli.error.cancelled('Not a named webtask.');
     }
-    var targetProfile = profile;
-    var claims = _(webtask).omit(['jti', 'iat', 'ca']).value();
-    var hasInlineCode = url.parse(webtask.url).protocol === 'webtask:';
-    if (hasInlineCode) {
-        delete claims.url;
-    } else {
-        delete claims.code;
-    }
+
+    var claims = cloneWebtaskData(data);
     claims.jtn = target.name || claims.jtn;
     claims.ten = target.container || claims.ten;
 
     var pendingCreate;
     if (target.profile) {
         pendingCreate = loadProfile(target.profile)
             .then(function(profile) {
-                targetProfile = profile;
+                target.profile = profile;
                 claims.ten = target.container || profile.container || claims.ten;
-                return targetProfile.createRaw(claims);
+                return target.profile.createRaw(claims);
             });
     } else {
-        pendingCreate = targetProfile.createRaw(claims);
+        target.profile = webtask.sandbox;
+        pendingCreate = target.profile.createRaw(claims);
     }
 
     return pendingCreate
         .then(function() {
```

Comment:
The `pendingCreate` promise will resolve with a `Webtask` instance corresponding to the `target`. This can be used below to facilitate some of the logic.